src/Symfony/Component/Security/Core/User/UserInterface.php

```diff
@@ -56,6 +56,8 @@ public function eraseCredentials(): void;
 
     /**
      * Returns the identifier for this user (e.g. username or email address).
+     *
+     * @return non-empty-string
```

Comment:
This is only true after the removal of the deprecation, right now our code can return an empty string

Reply:
I think this is fine as it does not break. We use phpdoc to introduce new parameters to a method for instance

Reply:
But when introducing new parameters we are in control of the code handling the additionally passed arguments (it's the code we are updating). For return types that's not true as we do not control the consuming code.

Reply:
Type annotations are declarative, not imperative. We should always have them describe the non-deprecated API IMHO.